holidays/ical.py

```diff
@@ -0,0 +1,222 @@
+                    parts.append(line[part_start:i])
+                    part_start = i
+                    part_len = char_byte_len + 1  # line start with space
+
+            parts.append(line[part_start:])
+            return CONTENT_LINE_DELIMITER_WRAP.join(parts)
+
+        # Return as-is if it doesn't exceed the limit
+        return line
+
+    def _generate_event(self, dt: date, holiday_name: str, holiday_length: int = 1) -> list[str]:
+        """
+        Generate a single holiday event.
+
+        :param date:
+            Holiday date.
+
+        :param holiday_name:
+            Holiday name.
+
+        :param holiday_length:
+            Holiday length in days, default to 1.
+
+        :return:
+            List of iCalendar format event lines.
+        """
+        # Escape special characters per RFC 5545.
+        sanitized_holiday_name = (
+            holiday_name.replace("\\", "\\\\").replace(",", "\\,").replace(":", "\\:")
+        )
+        event_uid = f"{uuid.uuid4()}@{self.holidays_version}.holidays.local"
+
+        lines = [
+            "BEGIN:VEVENT",
+            f"DTSTAMP:{self.ical_timestamp}",
+            f"UID:{event_uid}",
+            self._fold_line(f"SUMMARY:{sanitized_holiday_name}"),
+            f"DTSTART;VALUE=DATE:{dt:%Y%m%d}",
+            f"DURATION:P{holiday_length}D",
+            "END:VEVENT",
+        ]
+
+        return lines
+
+    def generate(self, return_bytes: bool = False) -> Union[str, bytes]:
+        """
+        Generate iCalendar data.
+
+        :param return_bytes:
+            If True, return bytes instead of string.
+
+        :return:
+            The complete iCalendar data
+            (string or UTF-8 bytes depending on return_bytes).
+        """
+        lines = [
+            "BEGIN:VCALENDAR",
+            f"PRODID:-//Vacanza//Open World Holidays Framework v{self.holidays_version}//"
+            f"{self.language}",
+            "VERSION:2.0",
+            "CALSCALE:GREGORIAN",
+        ]
+
+        sorted_dates = sorted(self.holidays.keys())
+        # Merged continuous holiday with the same name and use `DURATION` instead.
+        i = 0
+        while i < len(sorted_dates):
+            dt = sorted_dates[i]
+            names = self.holidays.get_list(dt)
+
+            for name in names:
+                days = 1
+                while (
+                    i + days < len(sorted_dates)
+                    and sorted_dates[i + days] == sorted_dates[i] + timedelta(days=days)
+                    and name in self.holidays.get_list(sorted_dates[i + days])
+                ):
+                    days += 1
+
+                lines.extend(self._generate_event(dt, name, days))
+
+            i += days
+
+        lines.append("END:VCALENDAR")
+        lines.append("")
+
+        output = CONTENT_LINE_DELIMITER.join(lines)
+        return output.encode() if return_bytes else output
+
+    def export_ics(self, filename: str = "calendar", export_path: Path = Path(".")) -> None:
```

Comment:
```suggestion
    def export_ics(self, file_path: str) -> None:
```
And let user just give the path (full or relative or just a name; .ics or .ical or something else). Not only `str` will work, but also `Path` or anything that works with `open()`.

Reply:
Implemented